### ISSUE TITLE
1.1.0 release breaks compatibility before Python 3.10

### ISSUE THREAD

**Issue by curtis-trynow-io** [theY4Kman/pytest-common-subject#4]
https://github.com/theY4Kman/pytest-common-subject/blob/master/CHANGELOG.md shows this info:

```
[1.1.0] — 2024-06-03
...
Changed
Drop support for Python 3.6 and 3.7 (both EOL) — minimum Python version is now 3.8
...
```

but this commit https://github.com/theY4Kman/pytest-common-subject/commit/1294141e38d79b811e91742d8e6e6a9367c3ef04#diff-e813a79ed7d9c9965f2859aeca250f36dbeb2ed37f06bf19827ff41a8170b537R186 here and in several other places introduces a pipe `|` union operator which is not supported before Python 3.10: https://docs.python.org/3/whatsnew/3.10.html PEP 604.

Effectively, all Python versions before 3.10 are now broken and will throw out:

```
  File "/usr/local/lib/python3.8/site-packages/pytest_common_subject/mixins.py", line 186, in AsyncCommonSubjectTestMixin
    def common_subject(self, *args) -> Callable[..., V | Awaitable[V]]:
TypeError: unsupported operand type(s) for |: 'TypeVar' and '_GenericAlias'
```

or similar as these lines are encountered.

Workaround is to pin down to `==1.0.6` but please either update the CHANGELOG or revert this syntax to `Union[...]` instead.

**Comment by theY4Kman** [comment]
Oh, man, that's embarrassing! So sorry about that. It's about time I add all the supported Python versions to the tox matrix. 

I'll have this sorted tonight. And I might go the route of adding `from __future__ import annotations` to the top of the file, to keep the pipes _and_ the compatibility... I rather like the pipes :P (And nothing needs the annotations at runtime.)

**Comment by theY4Kman** [comment]
Alright, I've expanded the tox matrix to cover all the Python and pytest versions supported, and fixed the `TypeError` for Python 3.8 and 3.9.

[Version 1.1.1](https://pypi.org/project/pytest-common-subject/1.1.1/) has been pushed to PyPI :)
